corehq/apps/geospatial/tests/test_views.py

```diff
@@ -335,23 +335,25 @@ def tearDownClass(cls):
         super().tearDownClass()
 
     def test_get_users_with_gps(self):
-        expected_output = {
-            'user_data': [
-                {
-                    'id': self.user_a.user_id,
-                    'username': self.user_a.raw_username,
-                    'gps_point': '12.34 45.67',
-                },
-                {
-                    'id': self.user_c.user_id,
-                    'username': self.user_c.raw_username,
-                    'gps_point': '45.67 12.34',
-                },
-            ],
+        expected_results = {
+            self.user_a.user_id: {
+                'id': self.user_a.user_id,
+                'username': self.user_a.raw_username,
+                'gps_point': '12.34 45.67',
+            },
+            self.user_c.user_id: {
+                'id': self.user_c.user_id,
+                'username': self.user_c.raw_username,
+                'gps_point': '45.67 12.34',
+            }
         }
         self.client.login(username=self.username, password=self.password)
         response = self.client.get(self.endpoint)
-        self.assertEqual(response.json(), expected_output)
+        response_json = response.json()
+        self.assertTrue('user_data' in response_json)
+        for user in response_json['user_data']:
+            user_id = user['id']
+            self.assertEqual(user, expected_results[user_id])
```

Comment:
These assertions will not fail if `response_json['user_data']` contains an incomplete set of results.
```suggestion
        user_data = {user['id']: user for user in response_json['user_data']}
        self.assertEqual(user_data, expected_results)
```

Reply:
Ah, good catch. I have changed this as per suggestion ([6532b87](https://github.com/dimagi/commcare-hq/pull/33981/commits/6532b87892ad20aac4828e246defcda922789fc9)).